howso/attributes.amlg

```diff
@@ -273,6 +273,39 @@
 			)
 		)
 
+		; check that ordinal features are either numbers or have a valid ordered list of allowed values
+		(if (size ordinals_map)
+			(let
+				(assoc invalid_ordinals (list))
+				(map
+					(lambda (let
+						(assoc
+							current_feature (current_index 1)
+							current_ordinal_map (get ordinals_map (current_index 1))
+						)
+						(if
+							(and
+								(!= "number" (get current_ordinal_map "data_type"))
+								(= (null) (get current_ordinal_map (list "bounds" "allowed")))
+							)
+							(accum (assoc invalid_ordinals current_feature))
+						)
+					))
+					ordinals_map
+				)
+				(if (> (size invalid_ordinals) 0)
+					(conclude (call !Return
+						(assoc errors
+							(list (concat
+								"The ordinal feature(s): " (apply "concat" (trunc (weave invalid_ordinals ", "))) " must either have a list of ordered allowed values set if they "
+								"are non numerics values or have the `data_type` attribute set to 'number' if it is a numeric value. To set the list of "
+								"ordered allowed values, you can either use the `ordinal_feature_values` parameter when calling `infer_feature_attributes`, "
```

Comment:
Do we want the Engine to directly reference things in the Python client? Personally, I would have guessed not, but not typically my decision

Reply:
yea... I mean its one of those where if I don't, it may not be clear how to do it in python